File: problem/Lexicographically_String/generateString.js
/**
 * @param {string} s
 * @param {string} t
 * @return {string}
 */
var generateString = function(s, t) {
    const n = s.length;
    const m = t.length;
    const len = n + m - 1;
    let ans = Array(len).fill('?');

    for (let i = 0; i < n; i++) {
        if (s[i] === 'T') {
            for (let j = 0; j < m; j++) {
                if (ans[i + j] !== '?' && ans[i + j] !== t[j]) {
                    return "";
                }
                ans[i + j] = t[j];
            }
        }
    }

    const old_ans = [...ans];
    for (let i = 0; i < len; i++) {
        if (ans[i] === '?') ans[i] = 'a';
    }

    for (let i = 0; i < n; i++) {
        if (s[i] === 'F') {
            let currentSub = ans.slice(i, i + m).join('');
            if (currentSub === t) {
                let ok = false;
                for (let j = i + m - 1; j >= i; j--) {
                    if (old_ans[j] === '?') {
                        ans[j] = 'b';
                        ok = true;
                        break;
                    }
                }
                if (!ok) return "";
            }
        }
    }

    return ans.join('');
};

File: problem/Lexicographically_String/generateString.py
class Solution:
    def generateString(self, s, t):
        n, m = len(s), len(t)
        ans = ['?'] * (n + m - 1)
        
        for i, b in enumerate(s):
            if b != 'T':
                continue

            for j, c in enumerate(t):
                v = ans[i + j]
                if v != '?' and v != c:
                    return ""
                ans[i + j] = c
        
        old_ans = ans
        ans = ['a' if c == '?' else c for c in ans] 
        
        for i, b in enumerate(s):
            if b != 'F':
                continue

            if ''.join(ans[i: i + m]) != t:
                continue

            for j in range(i + m - 1, i - 1, -1):
                if old_ans[j] == '?':
                    ans[j] = 'b'
                    break
            else:
                return ""
        return ''.join(ans)